File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
        int n = grid.size();
        vector<vector<int>> v(n-2, vector<int>(n-2,-1));
        for(int i=0; i<n; i++) {
            for(int j=0; j<n; j++) {
                for(int p=min(i, n-3); p>=max(i-2, 0); p--)
                    for(int q=min(j, n-3); q>=max(j-2, 0); q--)
                        v[p][q] = max(v[p][q], grid[i][j]);
            }
        }
        return v;
    }
};

File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.java
class Solution {
    public int[][] largestLocal(int[][] grid) {
        int n=grid.length;
        int[][] maxLocal=new int[n-2][n-2];
        
        
        for(int i=0;i<=grid.length-3;i++){
            
            for(int j=0;j<=grid.length-3;j++){
                maxLocal[i][j]=max(grid,i,j);
            }
            
        }
    
    return maxLocal;
    }
        
        public int max(int[][] grid,int i,int j){
            
            int mx=Integer.MIN_VALUE;
            
            for(int row=i;row<i+3;row++){
                for(int col=j;col<j+3;col++){
                    if(grid[row][col]>mx){
                        mx=grid[row][col];
                    }
                }
            }
            
            return mx;
        }
        
        
    }